frontend/src/metabase/dashboard/components/QuestionPicker/QuestionList.tsx

```diff
@@ -187,3 +194,35 @@ export function QuestionList({
     </>
   );
 }
+
+const VisualizerModalWithCardId = (
+  props: { cardId: CardId } & ComponentProps<typeof VisualizerModal>,
+) => {
+  const { cardId, ...otherProps } = props;
+  const [selectedCard, setSelectedCard] = useState<Card<DatasetQuery>>();
+  const dispatch = useDispatch();
+
+  const loadCard = useCallback(async () => {
+    const { data } = await dispatch(
+      cardApi.endpoints.getCard.initiate({ id: cardId }),
+    );
+
+    setSelectedCard(data);
+  }, [dispatch, cardId]);
+
+  useEffect(() => {
+    loadCard();
+  }, [loadCard]);
```

Comment:
We should use the `useGetCardQuery` hook here, please take a look at an [example](https://github.com/metabase/metabase/blob/0252fb2aa01929d5926f396752856923d3802fca/frontend/src/metabase/common/components/QuestionPicker/components/QuestionPicker.tsx#L65)